R/geom-abline.r

```diff
@@ -82,6 +82,16 @@ geom_abline <- function(mapping = NULL, data = NULL,
 
   # Act like an annotation
   if (!missing(slope) || !missing(intercept)) {
```

Comment:
I still think this code path should only be executed if no dataset is provided. It's not intuitive to overwrite a dataset the user has provided (and here this also happens without any kind of warning or message.) I can't envision any scenario in which replacing an explicitly provided dataset would be the right course of action.